components/Frame/HeaderNew.js

```diff
@@ -0,0 +1,387 @@
+  return isActiveRoute(active, 'index', {})
+}
+
+let routeChangeStarted
+
+const HeaderNew = ({
+  inNativeApp,
+  inNativeIOSApp,
+  isMember,
+  dark,
+  me,
+  t,
+  secondaryNav,
+  showSecondary,
+  router,
+  formatColor,
+  pullable = true,
+  hasOverviewNav,
+  children
+}) => {
+  const [isAnyNavExpanded, setIsAnyNavExpanded] = useState(false)
+  const [expandedNav, setExpandedNav] = useState(null)
+  const [isMobile, setIsMobile] = useState()
+  const [headerHeightState, setHeaderHeightState] = useState()
+  const [headerOffset, setHeaderOffset] = useState(0)
+  const [userNavExpanded, setUserNavExpanded] = useState(false)
+
+  const fixedRef = useRef()
+  const diff = useRef(0)
+  const lastY = useRef()
+  const headerHeight = useRef()
+  const lastDiff = useRef()
+
+  const textFill = dark ? colors.negative.text : colors.text
+  const logoFill = dark ? colors.logoDark || '#fff' : colors.logo || '#000'
+  const backButton = inNativeIOSApp && me && !isFront(router)
+
+  const toggleExpanded = target => {
+    if (target.id === expandedNav) {
+      setIsAnyNavExpanded(false)
+      setExpandedNav(null)
+    } else if (isAnyNavExpanded) {
+      setExpandedNav(target.id)
+    } else {
+      setIsAnyNavExpanded(!isAnyNavExpanded)
+      setExpandedNav(target.id)
+    }
+  }
+
+  const closeHandler = () => {
+    if (isAnyNavExpanded) {
+      setIsAnyNavExpanded(false)
+      setExpandedNav(null)
+      setUserNavExpanded(false)
+    }
+  }
+
+  const goTo = (pathName, route) => e => {
+    if (shouldIgnoreClick(e)) {
+      return
+    }
+    e.preventDefault()
+    if (router.pathname === pathName) {
+      window.scrollTo(0, 0)
+      closeHandler()
+    } else {
+      Router.pushRoute(route).then(() => window.scrollTo(0, 0))
+    }
+  }
+
+  const onScroll = () => {
+    const y = Math.max(window.pageYOffset, 0)
+
+    if (isAnyNavExpanded) {
+      diff.current = 0
+    } else {
+      const newDiff = lastY.current ? lastY.current - y : 0
+      diff.current += newDiff
+      diff.current = Math.min(Math.max(-headerHeight.current, diff.current), 0)
+    }
+
+    if (diff.current !== lastDiff.current) {
+      fixedRef.current.style.top = `${diff.current}px`
+      setHeaderOffset(diff.current)
+    }
+
+    lastY.current = y
+    lastDiff.current = diff.current
+  }
+
+  const measure = () => {
+    const mobile = window.innerWidth < mediaQueries.mBreakPoint
+    if (mobile !== isMobile) {
+      setIsMobile(mobile)
+    }
+    const { height } = fixedRef.current.getBoundingClientRect()
+    headerHeight.current = height
+    if (height !== headerHeightState) {
+      setHeaderHeightState(height)
+    }
+    onScroll()
+  }
+
+  useEffect(() => {
+    window.addEventListener('scroll', onScroll)
+    window.addEventListener('resize', measure)
+    measure()
+    return () => {
+      window.removeEventListener('scroll', onScroll)
+      window.removeEventListener('resize', measure)
+    }
+  }, [showSecondary])
```

Comment:
this effect does not correctly scope dependencies and the on scroll and reize functions might use outdated state

Reply:
moved onscroll and measure inside useeffect and added all dependencies